### ISSUE TITLE
Weird discontinuities in training

### ISSUE THREAD

**Issue by emilysilcock** [facebookresearch/DPR#205]
Hi there, 

Thanks for a fantastic repo. 

I'm running biencoder training with bart-large and training on the mnli dataset. 

To start with it's training pretty well, if slowly, but then I get a huge jump in training and evaluation loss after a couple of epochs. I can't for the life of me work out why this is, so I'm just after any suggestions of what could be causing this behaviour. My full config is below.

Any suggestions would be super helpful - TIA!

<img width="527" alt="image" src="https://user-images.githubusercontent.com/86472495/152552542-9ab2cce9-4f04-4410-9003-fb217fdda8b7.png">

```
# @package _group_

batch_size: 32
dev_batch_size: 128
adam_eps: 1e-8
adam_betas: (0.9, 0.999)
max_grad_norm: 2.0
log_batch_step: 1
train_rolling_loss_step: 100
weight_decay: 0.0
learning_rate: 7e-5

# Linear warmup over warmup_steps.
warmup_steps: 150

# Number of updates steps to accumulate before performing a backward/update pass.
gradient_accumulation_steps: 1

# Total number of training epochs to perform.
num_train_epochs: 40
eval_per_epoch: 4
hard_negatives: 1
other_negatives: 1
val_av_rank_hard_neg: 30
val_av_rank_other_neg: 30
val_av_rank_bsz: 128
val_av_rank_max_qs: 10000
```

**Comment by vlad-karpukhin** [comment]
Hi @emilysilcock ,
I'd recommend to try to play with learning_rate & max_grad_norm parameters to see if it prevents the model from divergence.
I actually haven't tried to train any 'large' models, it is just my guess.
Also, how many epochs are there in your training before the spike?

**Comment by emilysilcock** [comment]
Thanks so much! I hadn't really looked at the max_grad_norm, but small changes to that seem to make a big difference. I'm now training to 30+ epochs without divergence.
